### PR TITLE
Option to throw exceptions when an error is encountered

### PR DESCRIPTION
**Pull request by erayd**
Add option to throw exceptions when an error is encountered, rather than continuing.

### CODE REVIEW

**Comment by bighappyface** [comment]
@erayd please rebase and update the readme to reflect the new public `validate` method added in #351

**Comment by erayd** [comment]
@bighappyface Done :-).

File: README.md [hunk 1]
```diff
@@ -130,9 +139,23 @@ $jsonValidator = new Validator( new Factory($schemaStorage));
...
+$jsonValidator->validate($jsonToValidateObject, $jsonSchemaObject);
 ```
 
+### Configuration Options
+A number of flags are available to alter the behavior of the validator. These can be passed as the
+third argument to `Validator::validate()`, or can be provided as the third argument to
+`Factory::__construct()` if you wish to persist them across multiple `validate()` calls.
+
+| Flag | Description |
+|------|-------------|
```
Comment (shmax) [inline README.md]: Did you forget `CHECK_MODE_APPLY_DEFAULTS`?

**Comment by erayd** [comment]
@shmax No - I deliberately omitted it, because that feature isn't part of this pull request. Seemed more appropriate to add it in PR #349 after this one is merged.

**Comment by shmax** [comment]
Ah, gotcha. Ok. LGTM

**Pull request merged**